### ISSUE TITLE
[cloud_firestore] improve documentation for offline use

### ISSUE THREAD

**Issue by oodavid** [FirebaseExtended/flutterfire#2474]
**Describe the bug**
While offline, trying to `get` a firestore document that has not yet been created throws an error:

`PlatformException (PlatformException(Error performing get, Failed to get document because the client is offline., null))`

**To Reproduce**

```dart
final Firestore _db = Firestore.instance;
// Create a NEW document
DocumentReference docRef = _db.collection('jobs').document();
// Then try to get it
docRef.get().then((DocumentSnapshot snapshot){
   // ... error thrown
});
```

**Expected behavior**
The error should not be thrown as offline persistence is enabled by default

**Additional context**
The example above is trite. It is however, the smallest amount of code to reproduce the bug.

I personally saw that error while implementing a function that would create or update a document in slightly different ways depending on whether the document exists or not:

```dart
Future<void> upsertValues({
    @required DocumentReference docRef, 
    @required Map<String, dynamic> newValues,
}) {
    docRef.get().then((DocumentSnapshot snapshot){
        if(snapshot.exists){
            reference.updateData(...);
        } else {
            reference.setData(...);
        }
    });
}
```

**Flutter doctor**

```
[✓] Flutter (Channel stable, v1.12.13+hotfix.9, on Mac OS X 10.15.4 19E287, locale en-GB)
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 11.2.1)
[✓] Android Studio (version 3.5)
[✓] VS Code (version 1.44.2)
[✓] Connected device (1 available)
• No issues found!
```

**Package Version**

cloud_firestore: ^0.13.5

**Comment by oodavid** [comment]
I've been thinking about this, and I can see that it's not a trivial problem.

In my example, I'm using `.doc()` to _create_ a new document. Which is straightforward.
On the other hand `.doc('some-uid)` may or may not create a new document.

When offline, I can see that there's no way to rely on the response from `.get()`, this is made clear by trying to access `snapshot.exists`.

I assume that this *won't* get fixed, but a good resolution could be improved documentation and error messaging. If the error message pointed to documentation, that would be a huge boon for anyone experiencing this edge-case.

**Comment by Salakar** [comment]
Hey all, as part of our on-going work for #2582, this has been resolved in our Firebase Firestore rework (#2913) (which includes new documentation on offline usage) - which has now been merged into master. We'll look at publishing some prereleases in the next few days. Thank you
